R/generateFilterValues.R

```diff
@@ -125,21 +125,26 @@ generateFilterValuesData = function(task, method = "randomForestSRC_importance",
     }
 
   } else {
-    fval = lapply(filter, function(x) {
-      x = do.call(x$fun, c(list(task = task, nselect = nselect), more.args[[x$name]]))
-      missing.score = setdiff(fn, names(x))
-      x[missing.score] = NA_real_
-      x[match(fn, names(x))]
-    })
+    index_names = names(method)
+	if (is.null(index_names))
+		index_names = method
+	fval = mapply(function(x, name) {
+			x = do.call(x$fun, c(list(task = task, nselect = nselect), more.args[[name]]))
+			missing.score = setdiff(fn, names(x))
+			x[missing.score] = NA_real_
+			x[match(fn, names(x))]
+    }, filter, index_names, SIMPLIFY = FALSE)    
     fval = do.call(cbind, fval)
     colnames(fval) = method
-    types = vcapply(getTaskData(task, target.extra = TRUE)$data[fn], getClass1)
+	if (!is.null(names(method)))
+		colnames(fval) = names(method)
+	types = vcapply(getTaskData(task, target.extra = TRUE)$data[fn], getClass1)
 
     out = data.table(name = row.names(fval),
       type = types, fval, row.names = NULL, stringsAsFactors = FALSE)
 
     # variable.factor = FALSE has no effect
-    out = melt(out, value.name = "value", measure.vars = method,
+    out = melt(out, value.name = "value", measure.vars = colnames(fval),
       variable.name = "method")
```

Comment:
FYI: This will trigger a merge conflict after #2698 has been merged

Reply:
Yes it does.

Reply:
Yes it does. What can I do about this?

Reply:
Nothing, you cannot prevent a merge conflict ;) I just stated this because I was changing some code in this part in another PR, so I knew the conflict would appear.